storage-proofs/src/porep/stacked/vanilla/params.rs

```diff
@@ -504,17 +504,10 @@ impl<H: Hasher> Labels<H> {
     }
 
     pub fn verify_stores(&self, callback: VerifyCallback, cache_dir: &PathBuf) -> Result<()> {
-        for label in &self.labels {
-            callback(label, BINARY_ARITY)?;
-
-            let current_path = StoreConfig::data_path(&label.path, &label.id);
-            if !current_path.starts_with(&cache_dir) {
-                return Err(anyhow!(
-                    "Label path mis-match [label vs cache_dir: {:?}/{:?}]",
-                    current_path,
-                    cache_dir
-                ));
-            }
+        let updated_path_labels = self.labels.clone();
+        for mut label in updated_path_labels {
```

Comment:
there is already `fn update_root<P: AsRef<Path>>(&mut self, root: P)` down below for exactly this

Reply:
Yes, if we clone the entire `Labels` object, it could be re-used.  This seemed simpler in this case.

Reply:
no need to clone, you could just change this function signature to take `&mut self` and call `update_root` as the first step

Reply:
Ok.  We also update_root in the next phase.

Reply:
Or alternatively call `update_root` manually before calling this. Otherwise this gives a false sense of correctness, as this checks against a different cache path that the struct that the caller actually holds

Reply:
Just to be clear, the `validate_cache_for_precommit_phase2` will change signature to take a mutable `SealPreCommitPhase1Output`, correct?

Reply:
hmmmmmmmmm

Reply:
> Otherwise this gives a false sense of correctness, as this checks against a different cache path that the struct that the caller actually holds

Agreed, that's kind of the point of being able to set the cache_path though.  We do more or less ignore what's in the output, in favor of the updated path (e.g. in the next phase)

Reply:
Your call, I dislike all solutions proposed, pick what you think is the least bad.

Reply:
I also dislike the solutions, but if I'm understanding correctly, the phases are doing the right thing, but the validate is not because it is not respecting the updated cache path.  By being able to specify a path at all in the validate, it seems it should be used (always) in favor of whatever's stored, as the caller may have even locally moved the data.  We can chat about briefly on our call today, but I think as-is should be correct.  The previous code ensured that the cache path matched what was in the phase output, but we know that won't be the case when files have moved, so I'd say that was incorrect(ly based on non-moving assumptions).

Reply:
alright, lets ship it

Reply:
Not opposed, but would like to hear from @magik6k if it resolves the issue.  Just want to be sure I didn't miss anything.  If it's easier to test from master, let's ship.